File: mobile_ui/quickchat_ui/lib/features/user/domain/repositories/user_repository.dart
import 'package:quickchat_ui/core/logic/result.dart';
import 'package:quickchat_ui/features/authentication/data/dto/SearchData.dto.dart';
import 'package:quickchat_ui/features/authentication/data/dto/SearchUser.dto.dart';
import 'package:quickchat_ui/features/authentication/data/dto/UserData.dto.dart';

abstract class UserRepository {
  Future<Result<UserDataDto>> getUser(String userId);
  Future<Result<SearchDataDto>> search(SearchUserDto searchUserDto);
}

File: web_ui/app/features/user/domain/repositories/user_repository.ts
import { Result } from "@/app/core/result";
import { SearchDataDto } from "../../data/dto/SearchData.dto";
import { SearchUserDto } from "../../data/dto/SearchUser.dto";

export interface UserRepository {
  search(searchUserDto: SearchUserDto): Promise<Result<SearchDataDto>>;
}
